lib/cudnn/conv.jl

```diff
@@ -38,6 +38,9 @@ function ConvDesc(T, N, padding, stride, dilation, mode)
     else
         cudnnSetConvolutionNdDescriptor(cd[],N,cdsize(padding,N),cdsize(stride,N),cdsize(dilation,N),mode)
     end
+    if version() >= v"7"
+        cudnnSetConvolutionMathType(cd[], cudnnMathType_t(1))
```

Comment:
That's, CUDNN_TENSOR_OP_MATH, right? We can't use that unconditionally since it doesn't have the same accuracy. Instead: https://github.com/JuliaGPU/CUDA.jl/blob/463a41295bfede5125c584e6be9c51a4b9074e12/lib/cublas/CUBLAS.jl#L42-L46

Reply:
Yes, that is correct. I have added the extra conditions in this commit: https://github.com/JuliaGPU/CUDA.jl/pull/321/commits/480ea6fdb2edcca46152cd0398d7b6e456b748d7.
I have also added a condition for using the fast version of softmax over the accurate one.